File: insertion-sort/qtd_trocas.py
def ordenar(arr):
    contador = 0
    for i in range(1, len(arr)):
        temp = arr[i]
        j = i - 1

        while j >= 0 and arr[j] > temp:
            arr[j + 1] = arr[j]
            j-= 1
            contador += 1
        arr[j + 1] = temp
    return contador

arr = [3,1,5,4,9,7];

qtd = ordenar(arr)

print(qtd)

File: insertion-sort/qtd_trocas.c
#include <stdio.h>

int ordenar(int *arr, int n){
    int contador = 0;

    for(int i = 1; i < n; i++){
        int temp = arr[i];
        int j = i - 1;

        while(j >= 0 && arr[j] > temp){
            arr[j + 1] = arr[j];
            j--;
            contador++;
        }
        arr[j + 1] = temp;
    }
    return contador;
}

int main(){

    int arr[] = {3,1,5,4,9,7};

    int tamanho = sizeof(arr) / sizeof(arr[0]);

    int qtd = ordenar(arr, tamanho);

    printf("%d", qtd);

    return 0;
}